### PR TITLE
Api key docs

### PR DESCRIPTION
**Pull request by abhilash-kumar-nair**
This PR(https://modelonproducts.atlassian.net/browse/WAMS-9533) adds docs for JupyterHub auth

### CODE REVIEW

File: docs/source/setup.rst [hunk 1]
```diff
@@ -4,9 +4,38 @@ Setup
...
+safely.*
+
+If the token is lost, you can always generate a new one by clicking the "Request new API token"
+button.
+While initializing the client, you will be asked to enter the token key in a prompt.
+
+.. code-block:: pycon
+
+    >>>> from modelon.impact.client import Client
+    >>>> client = Client(url=impact_url)
```
Comment (efredriksson-modelon) [inline docs/source/setup.rst:27]: we should have `interactive=True` as you will only get a prompt if that is the case.

Comment (efredriksson-modelon) [reply docs/source/setup.rst:27]: Maybe we should have `url=<JupyterHub url>` to be consistent with the rest of the text?

File: docs/source/setup.rst [hunk 2]
```diff
@@ -4,9 +4,38 @@ Setup
...
+(located at <JupyterHub url>/token).You can get a secret token by clicking the 
+"Request new API token" button. These are tokens with full access to the JupyterHub API. 
+Anything you can do with JupyterHub can be done with these tokens.
+
+*Remember that this is the only time you will see the secret. Make sure to store it
+safely.*
+
+If the token is lost, you can always generate a new one by clicking the "Request new API token"
+button.
+While initializing the client, you will be asked to enter the token key in a prompt.
```
Comment (efredriksson-modelon) [inline docs/source/setup.rst:22]: token key -> JupyterHub API token

File: docs/source/setup.rst [hunk 3]
```diff
@@ -4,9 +4,38 @@ Setup
 This page describes the procedure to set up API keys to authenticate your login and
 various ways to use the generated API key to log in to the client.
 
+Authentication with JupyterHub
+******************************
+
+This section is ONLY applicable for users running Modelon Impact in a JupyterHub environment.
```
Comment (efredriksson-modelon) [inline docs/source/setup.rst:10]: Maybe we could add something like:
, otherwise skip to :ref:\`below\<Authentication with API keys\>\`.

**Review by efredriksson-modelon — CHANGES_REQUESTED** [review]
Nice with clear instructions on how to do this 👍

Added a few suggestions for clarification.

File: docs/source/setup.rst (hunk 1 above)
Comment (abhilash-kumar-nair) [reply docs/source/setup.rst:27]: The prompt doesn't require interactive flag set to true to show up?

Comment (efredriksson-modelon) [reply docs/source/setup.rst]: You should only get the prompt if `interactive=True`, that is the purpose of the interactive flag.

**Review by efredriksson-modelon — APPROVED** [review]
👍

**Pull request merged**